formatter/src/main/java/org/togetherjava/tjbot/formatter/Formatter.java

```diff
@@ -1,138 +1,31 @@
 package org.togetherjava.tjbot.formatter;
 
+import org.togetherjava.tjbot.formatter.formatting.CodeSectionFormatter;
 import org.togetherjava.tjbot.formatter.tokenizer.Lexer;
 import org.togetherjava.tjbot.formatter.tokenizer.Token;
-import org.togetherjava.tjbot.formatter.tokenizer.TokenType;
 
-import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /**
- * Formatter which can format a given string into a string which contains code blocks etc
+ * Formats code given as string. See {@link #format(CharSequence)}.
+ * <p>
+ * Best results are achieved for Java code.
  */
-public class Formatter {
-    /**
-     * Formats the given tokens
-     *
-     * @param tokens tokens to format
-     * @return resulting code
-     */
-    public String format(List<Token> tokens) {
-        List<Section> sections = sectionize(indexTokens(tokens));
-        StringBuilder result = new StringBuilder();
-
-        for (Section section : sections) {
-            if (section.isCodeSection()) {
-                result.append("```java\n")
-                    .append(writeCodeSection(section.tokens()))
-                    .append("\n```");
-            } else {
-                result.append(joinTokens(section.tokens()));
-            }
-        }
-
-        return result.toString();
-    }
-
-    /**
-     * Formats the given string using a given lexer
-     *
-     * @param input input to format
-     * @param lexer lexer to use
-     * @return resulting code
-     */
-    public String format(String input, Lexer lexer) {
-        return format(lexer.tokenize(input));
-    }
-
-    /**
-     * Joins given tokens together and normalizes whitespaces
-     *
-     * @param tokens tokens to join
-     * @return joined form of the tokens
-     */
-    private String joinTokens(List<Token> tokens) {
-        return tokens.stream().map(Token::content).collect(Collectors.joining());
-    }
-
-    /**
-     * Writes and formats a given code section (in form of a list of tokens) into a StringBuilder
-     * using a {@link CodeSectionFormatter}
-     *
-     * @param tokens tokens to write
-     * @return written code sections
-     */
-    private StringBuilder writeCodeSection(List<Token> tokens) {
-        CodeSectionFormatter formatter = new CodeSectionFormatter(tokens);
-
-        formatter.format();
-
-        return formatter.result();
-    }
+public final class Formatter {
+    private final Lexer lexer = new Lexer();
 
     /**
-     * Indexes tokens to contain information about whether they are code tokens or not
+     * Formats the given string.
+     * <p>
+     * Best results are achieved for Java code.
```

Comment:
"best results", that sounds like this works for JS, but not as good.

Instead just say "Only works with Java"

Reply:
> "best results", that sounds like this works for JS, but not as good.

but it does. and its expected to be used for that as well.

right now, the format action is available for all languages and yields okayish results for everything i tested (except languages without semicolons)